test/e2e/storage/utils/deployment.go

```diff
@@ -98,7 +98,10 @@ func PatchCSIDeployment(f *framework.Framework, o PatchCSIOptions, object interf
 		patchContainers(spec.Containers)
 		patchVolumes(spec.Volumes)
 		if o.NodeName != "" {
-			spec.NodeName = o.NodeName
+			// Don't directly set NodeName because it bypasses the scheduler and can
+			// cause kubelet to immediately Fail the pod if there's no space on the
+			// node.
+			spec.NodeSelector = map[string]string{v1.LabelHostname: o.NodeName}
```

Comment:
do we need to merge the selector? this is stomping unconditionally

Reply:
I think if someone configures a driver to only run on a specific node, then it should override any other existing selectors/affinity. I can have this return an error instead if it's already set.